client/src/components/Header/components/universal-nav.css

```diff
@@ -142,33 +151,148 @@
+}
+
+.display-menu .theme-disabled {
+  height: 4.2rem;
+}
+
+.nav-link-header label {
+  font-weight: normal;
+  margin-bottom: 0;
+}
+
+.nav-lang-menu {
+  display: none;
+  padding: 0;
+  color: var(--gray-90);
+  background-color: var(--gray-80);
   width: 100%;
-  border: none;
-  position: relative;
+  position: absolute;
   -webkit-appearance: none;
   -moz-appearance: none;
   appearance: none;
-  background-image: url('data:image/svg+xml;charset=UTF-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2212%22%20height%3D%2212%22%20viewBox%3D%220%200%2012%2012%22%3E%3Ctitle%3Edown-arrow%3C%2Ftitle%3E%3Cg%20fill%3D%22%23ffffff%22%3E%3Cpath%20d%3D%22M10.293%2C3.293%2C6%2C7.586%2C1.707%2C3.293A1%2C1%2C0%2C0%2C0%2C.293%2C4.707l5%2C5a1%2C1%2C0%2C0%2C0%2C1.414%2C0l5-5a1%2C1%2C0%2C1%2C0-1.414-1.414Z%22%20fill%3D%22%23ffffff%22%3E%3C%2Fpath%3E%3C%2Fg%3E%3C%2Fsvg%3E');
-  background-size: 10px;
-  background-position: calc(100% - 15px) center;
-  background-repeat: no-repeat;
+  list-style-type: none;
+  margin: 0;
+  border: 2px solid var(--gray-90);
+  border-top: 0;
+  border-bottom: 0;
+  top: 0;
+  overflow-y: auto;
+  scrollbar-width: auto;
+  /* lang menu should always be full height */
+  height: 19.9rem;
 }
 
-.nav-link-lang-dropdown:focus {
+/* main menu must be at least as tall as lang menu (when displayed) 
+   unless there isn't enough view port height */
+.nav-list.display-lang-menu {
+  min-height: min(19.9rem, calc(100vh - var(--header-height)));
+}
+
+.nav-lang-menu > button:hover {
+  background-image: none !important;
+}
+
+.nav-lang-menu .nav-link {
+  color: #fff;
+  background-color: inherit;
+  padding-top: 0;
+}
+
+.nav-lang-menu .nav-link:focus {
+  color: #000;
+  background-color: var(--gray-10);
   outline: none;
 }
 
+.nav-lang-menu .nav-link:focus:not(:focus-visible) {
+  background-color: inherit;
+}
+
+.nav-lang-menu .nav-link:hover {
+  background-color: var(--gray-10) !important;
+}
+
+.nav-lang-button {
+  display: flex;
+  justify-content: space-between;
+  background-image: none;
+}
+
+.nav-lang-button:hover svg,
+.nav-lang-button:focus {
+  fill: var(--gray-10);
+}
+
+.nav-lang-button:focus:not(:focus-visible) {
+  fill: revert;
+}
+
+.nav-lang-button[aria-expanded='true'] + .nav-lang-menu {
+  display: block;
+}
+
+.nav-lang-button svg {
+  height: 1rem;
+  width: 1rem;
+}
+
+.dark-palette .nav-lang-button:focus {
+  fill: revert;
+}
+
+.nav-lang-menu-option[aria-current='true'] {
+  /* check mark for current language */
+  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' width='32' height='32' preserveAspectRatio='xMidYMid meet' viewBox='0 0 16 16'%3E%3Cg fill='white'%3E%3Cpath d='M12.736 3.97a.733.733 0 0 1 1.047 0c.286.289.29.756.01 1.05L7.88 12.01a.733.733 0 0 1-1.065.02L3.217 8.384a.757.757 0 0 1 0-1.06a.733.733 0 0 1 1.047 0l3.052 3.093l5.4-6.425a.247.247 0 0 1 .02-.022z'%3E%3C/path%3E%3C/g%3E%3C/svg%3E");
```

Comment:
Ya, perhaps not my finest hour. I think I went with the CSS route because it was the least amount of work. No problem refactoring this but if we could do it in a separate PR that would be great.